flexget/plugins/input/telegram_input.py

```diff
@@ -0,0 +1,228 @@
+"""Plugin for telegram input"""
+import re
+from loguru import logger
+from flexget.config_schema import one_or_more
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+
+logger = logger.bind(name='telegram_input')
+
+_TELEGRAM_API_URL = "https://api.telegram.org/"
+
+
+class TelegramInput:
+    """
+    Parse any messages from Telegram and fills fields with regex
+
+    Example::
+
+      token: <token>
+      clean_update: yes
+      whitelist:
+        - username: <my_username>
+        - group: <my_group>
+        - fullname:
+            first: <my_first_name>
+            sur: <my_surname>
+      entry:
+        <field>: <regexp to match value>
+
+    Note: If not declated, title will be the message
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'token': {'type': 'string'},
+            'types' : one_or_more({'type': 'string', 'enum': ['private', 'group']}),
+            'whitelist': {
+                'type': 'array',
+                'minItems': 1,
+                'items': {
+                    'oneOf': [
+                        {
+                            'type': 'object',
+                            'properties': {'username': {'type': 'string'}},
+                            'required': ['username'],
+                            'additionalProperties': False,
+                        },
+                        {
+                            'type': 'object',
+                            'properties': {
+                                'fullname': {
+                                    'type': 'object',
+                                    'properties': {
+                                        'first': {'type': 'string'},
+                                        'sur': {'type': 'string'},
+                                    },
+                                    'required': ['first', 'sur'],
+                                    'additionalProperties': False,
+                                }
+                            },
+                            'required': ['fullname'],
+                            'additionalProperties': False,
+                        },
+                        {
+                            'type': 'object',
+                            'properties': {'group': {'type': 'string'}},
+                            'required': ['group'],
+                            'additionalProperties': False,
+                        },
+                    ]
+                },
+            },
+
+            'clean_update': {'type': 'boolean','default': True},
+            'entry': {
+                'type': 'object',
+                'properties': {
+                    'url': {'type': 'string', 'format': 'regex'},
+                    'title': {'type': 'string', 'format': 'regex'},
+                },
+                'additionalProperties': {'type': 'string', 'format': 'regex'},
+                'required': [],
```

Comment:
Empty `required` parameter is causing the test failure

Reply:
I'll remove it! Tks